### PR TITLE
Feature/issue#5 addspentmoneyandplaytime

### PR DESCRIPTION
**Pull request by chrisweber1408**
The user can add spent money and playtime to his game. He can change the games crated by him but not games created from admin.

### CODE REVIEW

File: backend/src/main/java/com/example/dontwastetomuch/game/GameController.java [hunk 1]
```diff
@@ -21,9 +21,19 @@ public class GameController {
...
+    
     @GetMapping
     public List<Game> getAllGames(){
         return gameService.getAllGames();
     }
+
+    @GetMapping("/{gameId}")
+    public Game getOneGame(@PathVariable String gameId){
+        return gameService.getOneGame(gameId);
+    }
```
Comment (Droggelbecher92) [inline backend/src/main/java/com/example/dontwastetomuch/game/GameController.java:33]: What will happen if no game is found? How your gonna tell your Frontend how to act?

File: backend/src/main/java/com/example/dontwastetomuch/game/GameService.java [hunk 2]
```diff
@@ -20,4 +20,12 @@ public void addAGame(Game game) {
...
         return gameRepo.findAll();
     }
+
+    public Game getOneGame(String gameId) {
+        return gameRepo.findById(gameId).orElseThrow();
+    }
+
+    public void editGame(Game game) {
+        gameRepo.save(game);
+    }
```
Comment (Droggelbecher92) [inline backend/src/main/java/com/example/dontwastetomuch/game/GameService.java:30]: There is an unique index in your Game model. How will you handle duplicates?

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -12,6 +13,7 @@ export default function App() {
                 <Route path={"/login"} element={<WelcomePage/>}/>
                 <Route path={"/register"} element={<RegisterPage/>}/>
                 <Route path={"/"} element={<MainPage/>}/>
+                <Route path={"/:id"} element={<InfoPage/>}/>
```
Comment (Droggelbecher92) [inline frontend/src/App.tsx:16]: here you'll get into trouble later on. What will happen if the id is `register`? 
```suggestion
                <Route path={"/login"} element={<WelcomePage/>}/>
                <Route path={"/register"} element={<RegisterPage/>}/>
                <Route path={"/"} element={<MainPage/>}/>
                <Route path={"/infos/:id"} element={<InfoPage/>}/>
```

File: frontend/src/Pages/InfoPage.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                .then(response => response.data)
+                .then(data => {
+                    setGame(data)
+                    setGameName(data.gameName)
+                    setSpentMoney(data.spentMoney)
+                    setPlaytime(data.playtime)
+                    setApproved(data.approved)
+                })
+        }
+    },[id])
```
Comment (Droggelbecher92) [inline frontend/src/Pages/InfoPage.tsx:31]: Here you should check if a game was found

**Review by Droggelbecher92 — APPROVED** [review]
Would recommend to fix your Error-handling before adding new Logic. Maybe now is a good time for your first Integration test and some edge-case Unit-Tests 👍
